### DISCUSSION TITLE
How to compile the Development Version

### DISCUSSION THREAD

**Discussion by Pewibe** [arendst/Tasmota#12882] [General]
I have to build a Tasmota binary with at least V 9.5.0.2 or even the latest one. Until today I used only release versions for compiling. This is easy because a source file with the whole module tree is provided. For a dev version I miss such a source file. 

In the doc, chapter "compiling" I didn't found the answer. Be patient with my missing knowledge and thanks for your help in advance.

**Comment by barbudor** [comment]
If you are talking about the ZIP file of the whole tasmota tree, you can download from github either master (== release) or development. Just select the one you want from the branch selection dropbox.  You can also download a given tag or commit. 

The best would be to learn how to use git either command line or from within Visual Code

**Comment by ascillato** [comment]
All the info is in the docs at https://tasmota.github.io/docs/Compile-your-build/

By default, when you go to ( https://github.com/arendst/Tasmota ) you are already in the DEVELOPMENT BRANCH.

So, there you can use any of the following options:

1- The download button in Github, unzip and run Visual Studio Code with Platformio to that folder:
![image](https://user-images.githubusercontent.com/35405447/129459338-e86855d4-d99a-4d6d-9d6e-bf43de445448.png)


2- Or open it in Gitpod as the docs (that after logging to Gitpod it is just clicking as the following image):
![image](https://user-images.githubusercontent.com/35405447/129459399-5753ff92-0a77-45e7-89c2-8a0535f30cb2.png)


3- Or use Github Desktop Application:
![image](https://user-images.githubusercontent.com/35405447/129459422-4c71e12c-fea6-47f2-91f2-779e3fe740f1.png)
Then in Github Desktop click on REPOSITORY
![image](https://user-images.githubusercontent.com/35405447/129459489-c1b01cd9-f9dc-425b-92a7-c1ffe77b5e32.png)
Then on OPEN VISUAL STUDIO CODE
![image](https://user-images.githubusercontent.com/35405447/129459500-3c8094bc-d68e-4e2f-967a-bef7ad31bc66.png)
Then in VS CODE click on compile and you are done.
![image](https://user-images.githubusercontent.com/35405447/129459608-a5869acf-0745-44a3-98db-cbef7d59cda7.png)
